codec.go

```diff
@@ -78,24 +84,27 @@ const kDecompressorBufferSize = 8 * 1024
 
 // A 'deflate' decompression context for BLIP messages.
 type decompressor struct {
+	context   LogContext
```

Comment:
I think this should be renamed from `context` -> `logContext`, to make way for other types of contexts if need be (eg, what if this later needs to have a `securityContext`?  in that case it would be confusing to have `decompressor.context` be related to logging context)

The other reason to avoid plain old context is to avoid people getting confused with https://golang.org/pkg/context/.  In Google codebases it's common practice to pass in `ctx context.Context` as the first parameter to function/method calls

I realize the name is longer, so maybe `logCtxt` to save typing?

Also taking a step back on the pluggable logging, I think there could be an argument to be made for a single package level logger callback, similar to the way gocb does it:

https://godoc.org/github.com/couchbase/gocb#SetLogger

That way things like the `decompressor` wouldn't need an additional field (with the downside that it couldn't track local state as easily using a package level logger)

Reply:
I don't disagree with anything you're saying, but I feel that it's sort of overkill. Is it really worth another round of changes right now?

Reply:
I think if we don't change it at this point, it will probably never get changed.   Anyway I went ahead and changed it, but looks like you beat me to it since my push failed.